File: week1/3/A.java
import java.util.Set;
import java.util.stream.Collectors;

class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        int answer= 0;

        Set<Character> jewelsSet= jewels.chars().mapToObj(
                c -> (char) c
        ).collect(Collectors.toSet());

        for(char stone: stones.toCharArray()){
            if (jewelsSet.contains(stone)) ++answer;
        }

        return answer;
    }

    public static void main(String[] args) {
        Solution solution=new Solution();
        var result= solution.numJewelsInStones("aA", "aAAbbbb");
        System.out.println(result);
    }
}

File: week1/3/A.py
# https://leetcode.com/problems/jewels-and-stones/

class Solution:
    def numJewelsInStones(jewels: str, stones: str) -> int:
        # Allocate dictionary
        freqs= {}
        count= 0

        for stone in stones:
            if stone not in freqs:
                # allocate value 1
                freqs[stone]= 1
            else:
                # stone in freqs
                freqs[stone] += 1
        
        for jewel in jewels:
            if jewel in freqs:
                count+= freqs[jewel]

        return count
    
result= Solution.numJewelsInStones("aA", "aAAbbbb")
# Output: 3
print(result)